File: java/basics/src/main/java/utils/SubscriptionService.java
package utils;

import dev.restate.sdk.ObjectContext;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.VirtualObject;

@VirtualObject
public class SubscriptionService {

    @Handler
    public String create(ObjectContext ctx, String userId) {
        // Implementation here
        return "SUCCESS";
    }

    @Handler
    public void cancel(ObjectContext ctx) {
        System.out.println("Cancelling all subscriptions for user " + ctx.key());
    }
}

File: kotlin/basics/src/main/kotlin/utils/SubscriptionService.kt
package utils

import dev.restate.sdk.annotation.Handler
import dev.restate.sdk.annotation.VirtualObject
import dev.restate.sdk.kotlin.*

@VirtualObject
class SubscriptionService {

    @Handler
    fun create(userId: String): String =
        "SUCCESS"

    @Handler
    suspend fun cancel() =
        println("Cancelling all subscriptions for user ${objectKey()}")
}
